core/trino-spi/src/main/java/io/trino/spi/type/Decimals.java

```diff
@@ -287,6 +289,36 @@ public static BigDecimal rescale(BigDecimal value, DecimalType type)
         return value;
     }
 
+    public static long roundDown(Long value, int fromScale, int toScale)
```

Comment:
Let's not have the method specific to `roundDown`. Instead, let's have a method that can do rounding optionally

Reply:
What are your thoughts regarding the overflow check? What should we do for them? They are essential to be there in my opinion but like you rightly said that sometimes the reader just knows that the values won't overflow. Should we make that optional too?

I have a slightly more performant version (in my opinion) for this particular overflow check for example :
```java
if (toScale - fromScale < MAX_SHORT_PRECISION && value < logTenToNth(MAX_SHORT_PRECISION - (toScale - fromScale)) {
// throw exception
}
return value * logTenToNth(toScale - fromScale);
```
We don't calculate the result if it overflows, here.

Reply:
let's keep overflow checks

Reply:
Let me know if any other changes are required, @sopel39